spanner/cloud-client/backup_sample_test.py

```diff
@@ -0,0 +1,107 @@
+# limitations under the License.
+
+from google.cloud import spanner
+import pytest
+import random
+import string
+
+import backup_sample
+
+
+def unique_instance_id():
+    """ Creates a unique id for the database. """
+    return 'test-instance-{}'.format(''.join(random.choice(
+        string.ascii_lowercase + string.digits) for _ in range(5)))
+
+
+def unique_database_id():
+    """ Creates a unique id for the database. """
+    return 'test-db-{}'.format(''.join(random.choice(
+        string.ascii_lowercase + string.digits) for _ in range(5)))
+
+
+def unique_backup_id():
+    """ Creates a unique id for the backup. """
+    return 'test-backup-{}'.format(''.join(random.choice(
+        string.ascii_lowercase + string.digits) for _ in range(5)))
+
+
+INSTANCE_ID = unique_instance_id()
+DATABASE_ID = unique_database_id()
+RESTORE_DB_ID = unique_database_id()
+BACKUP_ID = unique_backup_id()
+
+
+@pytest.fixture(scope='module')
+def spanner_instance():
+    spanner_client = spanner.Client()
+    instance_config = '{}/instanceConfigs/{}'.format(
+        spanner_client.project_name, 'regional-us-central1')
+    instance = spanner_client.instance(INSTANCE_ID, instance_config)
+    op = instance.create()
+    op.result(30)  # block until completion
+    yield instance
+    instance.delete()
+
+
+@pytest.fixture(scope='module')
+def database(spanner_instance):
+    """ Creates a temporary database that is removed after testing. """
+    db = spanner_instance.database(DATABASE_ID)
+    db.create()
+    yield db
+    db.drop()
+
+
+def test_create_backup(capsys, database):
+    backup_sample.create_backup(INSTANCE_ID, DATABASE_ID, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert "Backup " in out
+    assert (BACKUP_ID + " of size 0 bytes was created at ") in out
+
+
+def test_restore_database(capsys):
+    backup_sample.restore_database(INSTANCE_ID, RESTORE_DB_ID, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert "Database " in out
+    assert (DATABASE_ID + " restored to ") in out
+    assert (RESTORE_DB_ID + " from backup ") in out
+    assert (BACKUP_ID + ".") in out
+
+
+def test_update_backup(capsys):
+    backup_sample.update_backup(INSTANCE_ID, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert "Backup " in out
+    assert (BACKUP_ID + " expire time was updated from ") in out
+    assert " to " in out
+
+
+def test_delete_backup(capsys, spanner_instance):
+    backup_sample.delete_backup(INSTANCE_ID, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert "Backup " in out
+    assert (BACKUP_ID + " has been deleted.") in out
+
+
+def test_cancel_backup(capsys):
+    backup_sample.cancel_backup(INSTANCE_ID, DATABASE_ID, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    cancel_success = "Backup creation was successfully cancelled." in out
+    cancel_failure = (
+        ("Backup was created before the cancel completed." in out) and
+        ("Backup deleted." in out)
+    )
+    assert cancel_success or cancel_failure
```

Comment:
We don't have tests for the list samples. Is that intentional?

Reply:
Yes. Originally it was because I was worried about them being flaky. Now that the tests are run in a new instance it makes sense to have them again. I'll add them tomorrow.